src/pages/docs/ai-transport/features/token-streaming/message-per-response.mdx

```diff
@@ -0,0 +1,412 @@
+---
+title: Message per response
+meta_description: "Stream individual tokens from AI models into a single message over Ably."
+---
+
+Token streaming with message-per-response is a pattern where every token generated by your model is appended to a single Ably message. Each complete AI response then appears as one message in the channel history while delivering live tokens in realtime. This uses [Ably Pub/Sub](/docs/basics) for realtime communication between agents and clients.
+
+This pattern is useful for chat-style applications where you want each complete AI response stored as a single message in history, making it easy to retrieve and display multi-response conversation history. Each agent response becomes a single message that grows as tokens are appended, allowing clients joining mid-stream to catch up efficiently without processing thousands of individual tokens.
+
+## How it works <a id="how-it-works"/>
+
+1. **Initial message**: When an agent response begins, publish an initial message with `message.create` action to the Ably channel with an empty or the first token as content.
+2. **Token streaming**: Append subsequent tokens to the original message by publishing those tokens with the `message.append` action.
+3. **Live delivery**: Clients subscribed to the channel receive each appended token in realtime, allowing them to progressively render the response.
+4. **Compacted history**: The channel history contains only one message per agent response, which includes all tokens appended to it concatenated together.
+
+You do not need to mark the message or token stream as completed; the final message content will automatically include the full response constructed from all appended tokens.
+
+<Aside data-type="important">
+Standard Ably message [size limits](/docs/platform/pricing/limits#message) apply to the complete concatenated message. The system validates size limits before accepting append operations. If appending a token would exceed the maximum message size, the append is rejected.
+</Aside>
+
+## Enable appends <a id="enable"/>
+
+Message append functionality requires the "Message annotations, updates, and deletes" [channel rule](/docs/channels#rules) enabled for your channel or [namespace](/docs/channels#namespaces).
```

Comment:
I don't think the use of the term "rule" and "namespace" is correct. It's using the term "rule" to refer to a single configurable attribute of a namespace, whereas I think "rule" **is** the namespace definition (comprising the settings for all of the configurable attributes).
I think a more appropriate statement here would be:

> Message append functionality requires "Message annotations, updates, and deletes" to be enabled in a [channel rule]() associated with the channel.